homeassistant/components/immich/update.py

```diff
@@ -0,0 +1,61 @@
+"""Update platform for the Immich integration."""
+
+from __future__ import annotations
+
+from awesomeversion import AwesomeVersion
+
+from homeassistant.components.update import UpdateEntity, UpdateEntityDescription
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+
+from .coordinator import ImmichConfigEntry, ImmichDataUpdateCoordinator
+from .entity import ImmichEntity
+
+# Coordinator is used to centralize the data updates
+PARALLEL_UPDATES = 0
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ImmichConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Add immich server update entity."""
+    coordinator = entry.runtime_data
+
+    min_version = AwesomeVersion("v1.134.0")
+    if AwesomeVersion(coordinator.data.server_about.version) >= min_version:
+        async_add_entities([ImmichUpdateEntity(coordinator)])
+
+
+class ImmichUpdateEntity(ImmichEntity, UpdateEntity):
+    """Define Immich update entity."""
+
+    def __init__(
+        self,
+        coordinator: ImmichDataUpdateCoordinator,
+    ) -> None:
+        """Initialize."""
+        super().__init__(coordinator)
+        self._attr_unique_id = f"{coordinator.config_entry.unique_id}_update"
+        self.entity_description = UpdateEntityDescription(
+            key="update",
+            translation_key="update",
+        )
```

Comment:
Why do we set an entity description?

Reply:
Why not? :grimacing: With this all needed arguments are set. Or do you mean to just set the short hand `_attr_* ` directly?

Reply:
you're right ... we don't need the `key` attribut :+1: